R/utils.R

```diff
@@ -19,28 +19,40 @@ deparse_1 <- function(expr, collapse = " ", width.cutoff = 500L, ...) {
 }
 
 add_colours_missing_names <- function(names, colour_aliases) {
+  # add colours to names differing by letter case
   names <- unique(names)
-  missing_names <- setdiff(names, names(colour_aliases))
-  all_colours <- colour_aliases[names(colour_aliases) == ""]
-  available_colours <- setdiff(
-    all_colours,
-    colour_aliases[names]
-  )
-  if (length(missing_names) <= length(available_colours)) {
-    missing_aliases <- setNames(available_colours[1:length(missing_names)], missing_names)
-    colour_aliases <- c(colour_aliases, missing_aliases)
-    inform(glue(
-      "Assigning colours to unrecognised names in data: {toString(missing_names)}."
-    ))
-  } else {
-    abort(
-      c(
-        glue("Not enough colours for plotting the data in the chosen colour palette."),
-        x = glue("Number of unique colours in the palette: {length(available_colours)}.
-                 You passed a column with {length(names)} distinct values."),
-        i = "Consider splitting over multiple plots."
+  names_lower <- unique(tolower(names))
+  missing_names_core <- setdiff(names_lower, names(colour_aliases))
+  missing_names_lettercase <- setdiff(names, names(colour_aliases))
+  if (length(missing_names_core) < length(missing_names_lettercase)) {
+    names_case_difference <- setdiff(missing_names_lettercase, missing_names_core)
+    aliases_case_difference <- c()
+    for (i in 1:length(names_case_difference)) {
```

Comment:
`1:lengh(x)` is dangerous. When `x` has length 0 the result isn't what you expect:

``` r
x <- character(0)
1:length(x)
#> [1] 1 0
```

<sup>Created on 2022-03-11 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Instead use with `seq_along(x)`. You may run `goodpractice::gp()` to detect more instances of this.